packages/studio/src/components/ComponentNode.tsx

```diff
@@ -109,19 +132,27 @@ export default function ComponentNode(props: ComponentNodeProps): JSX.Element {
   );
 }
 
-function useDeleteKeyListener(
-  isActiveComponent: boolean,
-  componentStateUUID: string
-) {
-  const removeComponent = useStudioStore((store) => {
-    return store.actions.removeComponent;
-  });
+function useDeleteKeyListener(isSelectedComponent: boolean) {
+  const [removeComponent, selectedComponentUUIDs, clearSelectedComponents] =
+    useStudioStore((store) => {
+      return [
+        store.actions.removeComponent,
+        store.pages.selectedComponentUUIDs,
+        store.pages.clearSelectedComponents,
+      ];
+    });
   return useCallback(
     (event: KeyboardEvent<HTMLDivElement>) => {
-      if (isActiveComponent && event.key === "Backspace") {
-        removeComponent(componentStateUUID);
+      if (isSelectedComponent && event.key === "Backspace") {
+        selectedComponentUUIDs.forEach((uuid) => removeComponent(uuid));
```

Comment:
I think it would make sense to have the `removeComponent` action remove the component from the selected components, just like it handles resetting the `activeComponentUUID` if the active component is deleted. That way, we shouldn't need to separately call `clearSelectedComponents` below

Reply:
Done